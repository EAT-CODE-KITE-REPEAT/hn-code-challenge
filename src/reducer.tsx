export type Story = {
  id: number;
  title: string;
  url: string;
  by: string;
  score: number;
  time: number;
  type: string;
  kids: number[];
  descendants: number;
};

export type Store = {
  stories: { storiesById: Object };
  allStories: string[];
};

export type ReduxAction = {
  type: string;

  // #toRemember put different values under different keys, to make clear the type
  story?: Story;
  pending: boolean;
  error: any;
};

const initReducer: Store = {
  stories: {
    storiesById: {}
  },
  allStories: []
};

export const reducer = (
  state: Store = initReducer,
  action: ReduxAction
): Store => {
  switch (action.type) {
    case "GET_TOP_STORIES": {
      return {
        ...state,
        allStories: action.allStories
      };
    }

    case "SET_STORY": {
      const stories = state.stories.storiesById || {};
      return {
        ...state,
        stories: {
          storiesById: { ...stories, [action.story.id]: action.story }
          //   stories: stories.concat([action.story])
        }
      };
    }

    case "PURGE": {
      return {
        ...initReducer
      };
    }

    default:
      return state;
  }
};
