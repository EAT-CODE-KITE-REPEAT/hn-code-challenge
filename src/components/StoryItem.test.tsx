import React from "react";
import renderer from "react-test-renderer";

import App from "./StoryItem";
import { Story } from "../reducer";

//dummy story to test with
const story: Story = {
  by: "TestUser",
  descendants: 1,
  id: 1,
  kids: [],
  score: 10,
  time: Date.now(),
  title: "Test",
  type: "story",
  url: "https://google.nl/"
};

describe("Index runs", () => {
  //snapshot test
  it("renders correctly", () => {
    const tree = renderer
      .create(<App index={1} page={1} story={story} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  //unit test
  it("has 1 child", () => {
    const tree = renderer
      .create(<App index={1} page={1} story={story} />)
      .toJSON();
    expect(tree.children.length).toBe(2);
  });
});
