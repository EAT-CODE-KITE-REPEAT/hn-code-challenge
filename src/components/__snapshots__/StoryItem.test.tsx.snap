// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index runs renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      Object {
        "flex": 1,
        "margin": 5,
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#AAA",
        "marginRight": 10,
        "width": 30,
      }
    }
  >
    2
    .
  </Text>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Object {
            "width": 700,
          }
        }
      >
        Test
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 10,
          }
        }
      >
        10
         by 
        TestUser
      </Text>
    </View>
  </View>
</View>
`;
